tools/scripts/start-local-registry.ts

```diff
@@ -2,35 +2,60 @@
  * This script starts a local registry for e2e testing purposes.
  * It is meant to be called in jest's globalSetup.
  */
-import { execFileSync, execSync, spawn } from 'child_process';
+import { execSync, spawn } from 'node:child_process';
+import { join } from 'node:path';
+import { releasePublish, releaseVersion } from 'nx/release';
 
 export default async () => {
   // local registry target to run
   const localRegistryTarget = '@code-pushup/cli-source:local-registry';
   // storage folder for the local registry
-  const storage = './tmp/local-registry/storage';
+  const storage = join(
+    'tmp',
+    'local-registry',
+    'storage',
+    process.env.NX_TASK_TARGET_PROJECT ?? '',
+  );
 
-  global.stopLocalRegistry = await startLocalRegistry({
+  // using storage
+  const { registry, stop } = await startLocalRegistry({
     localRegistryTarget,
     storage,
     verbose: true,
   });
 
-  // is is also possible to use nx release to publish the packages to the local registry
-  execFileSync(
-    'npx',
-    [
-      'nx',
-      'run-many',
-      '--targets',
-      'publish',
-      '--ver',
-      '1.0.0',
-      '--tag',
-      'e2e',
-    ],
-    { env: process.env, stdio: 'inherit', shell: true },
-  );
+  if (!registry) {
+    console.error('Local registry not started');
+    process.exit(1);
+  }
+  global.stopLocalRegistry = stop;
+  global.registry = registry;
+  try {
+    const version = execSync('git describe --tags --abbrev=0')
+      .toString()
+      .trim();
```

Comment:
this enables getting the latest version of the packages (since versioning is tag-based and unique across all projects.)